File: Python/learn/quickStart.py
from dotenv import load_dotenv
load_dotenv()

from langchain.agents import create_agent
from langchain.chat_models import init_chat_model
from langgraph.checkpoint.memory import InMemorySaver
from langchain.tools import tool
from tools.baidu_search import BaiduSearchTool
import prompts



baiduTool = BaiduSearchTool()

from datetime import datetime, timezone

@tool("current_time", description="获取当前年月日、时分秒")
def current_time(fmt: str = "%Y-%m-%d %H:%M:%S", tz: str = "local") -> str:
    """获取当前时间。
    可选参数：
    - fmt: 时间格式化字符串，默认 "%Y-%m-%d %H:%M:%S"
    - tz: 时区，"UTC" 或 "local"，默认 "local"
    返回格式化后的当前时间字符串。
    """
    try:
        print("获取当前时间")
        if tz.upper() == "UTC":
            now = datetime.now(timezone.utc)
        else:
            now = datetime.now()
        return now.strftime(fmt)
    except Exception:
        return datetime.now().isoformat()

# temperature：用于控制随机性的模型温度。
# max_tokens：输出令牌的最大数量。
# timeout：等待响应的最长时间（以秒为单位）。
# max_retries：失败请求的最大重试次数。
# base_url：自定义 API 端点 URL。
# rate_limiter：BaseRateLimiter控制请求率的实例。

model = init_chat_model(
    model="deepseek-chat",
    model_provider="deepseek",
    temperature=0.7,
    max_tokens=None,
    timeout=None,
    max_retries=2,
)

agent = create_agent(
    tools=[baiduTool, current_time],
    model=model,
    system_prompt=prompts.default_rag,
    checkpointer=InMemorySaver(),
)

# 保存工作流图表
import os
try:
    # 构建图片保存路径
    file_name = "create_agent.png"
    current_dir = os.path.dirname(os.path.abspath(__file__))
    image_path = os.path.join(current_dir, "images", file_name)
    # 检查文件是否已存在
    if os.path.exists(image_path):
        print(f"图表文件已存在")
    else:
        png_data = agent.get_graph().draw_mermaid_png()
        with open(image_path, "wb") as f:
            f.write(png_data)
        print(f"图表已保存为: {image_path}")
except Exception as e:
    print(f"保存失败: {e}")

# result = agent.invoke({"messages": [{"role": "user", "content": "深圳今天天气怎么样？适合什么穿搭？"}]})
# print(result["messages"][-1].content)
thread_id = 'default_thread_123'
while True:
    try:
        user_input = input("User: ")
        if (user_input.lower() in ['quit', 'exit', 'bye']):
            print("Goodbye!")
            break
        result = agent.invoke(
            {"messages": [{"role": "user", "content": user_input}]}, 
            {"configurable": {"thread_id": thread_id}}
            )
        print(result["messages"][-1].content)
    except :
        break

File: TypeScript/learn/agents/quick_start.ts
import { createAgent } from "langchain";
import { ChatDeepSeek } from "@langchain/deepseek";
import { MemorySaver } from "@langchain/langgraph";
import { BaiduSearchTool } from "../tools/baiduSearchTool";
import { getCurrentTimeTool } from "../tools/common";
import { HumanMessage } from "@langchain/core/messages";
const baiduTool = new BaiduSearchTool();

const systemPrompt = `
# 角色
你是一个专业的智能助手，具备强大的信息检索和生成能力。

# 核心要求
1. **信息检索**：使用搜索baidu_search工具获取最新、权威的客观信息和知识
2. **信息时效性**：请时刻使用get_current_time工具获取当前时间，确保信息时效性
3. **信息准确性**：生成内容请以搜索结果为准，避免臆测和幻觉，确保信息准确性
`;
const model = new ChatDeepSeek({
	model: "deepseek-chat",
	temperature: 0.8,
});

const checkpointer = new MemorySaver();

const agent = createAgent({
	model: model,
	tools: [baiduTool, getCurrentTimeTool],
	systemPrompt: systemPrompt,
	checkpointer,
});

const config = {
	configurable: { thread_id: "1" },
	context: { user_id: "1" },
};

// const invoke = async () => {
// 	const result = await agent.invoke(
// 		{
// 			messages: [{ role: "user", content: "请帮我写一个关于机器学习的博客" }],
// 		},
// 		config
// 	);
// 	if (result && result.messages && result.messages.length > 0) {
// 		console.log(result.messages[result.messages.length - 1].content);
// 	}
// };
// invoke();

// 交互式CLI：模仿 Python while True 输入循环
import * as readLine from "node:readline";
async function startCli(): Promise<void> {
	const rl = readLine.createInterface({
		input: process.stdin,
		output: process.stdout,
		terminal: true,
	});

	console.log("请输入搜索关键词，输入 quit/exit/bye 退出");
	rl.setPrompt("User: ");
	rl.prompt();

	rl.on("line", async (line: string) => {
		const userInput = line.trim();
		if (["quit", "exit", "bye"].includes(userInput.toLowerCase())) {
			console.log("Goodbye!");
			rl.close();
			return;
		}
		try {
			const result = await agent.invoke(
				{
					messages: [new HumanMessage({ content: userInput })],
				},
				config
			);
			// console.log(result);
			if (result && result.messages && result.messages.length > 0) {
				console.log(result.messages[result.messages.length - 1].content);
			}
		} catch (err: any) {
			console.log(`'${userInput}' 调用失败：${err?.message ?? String(err)}`);
		}
		rl.prompt();
	});

	rl.on("close", () => {
		// 与 Python 示例一致，退出时结束进程
		if (typeof process !== "undefined") {
			process.exit(0);
		}
	});
}
startCli();
